AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
+static const az_span command_empty_payload_maxmin_span = AZ_SPAN_LITERAL_FROM_STR("Payload Empty. Please specify parameter 'since'");
+
+// IoT Plug and Play Device Values for telemetry
+
+static int32_t device_current_temp;
+static int32_t device_current_light;
 static int32_t device_temperature_avg_total;
 static uint32_t device_temperature_avg_count = 0;
-static int32_t device_max_temp;
-static int32_t device_min_temp;
-static int32_t device_avg_temp;
+static int32_t device_max_temp = 0;
+static int32_t device_min_temp = INT32_MAX;
+static int32_t device_avg_temp = 0;
+
+/**************************************
+ A task to reboot (reset) the device
+static uint32_t reboot_task(void* payload)
+*************************************/
+static uint32_t reboot_task(void* payload)
+{
+	asm("RESET"); // never returns
+	return 0;
+}
 
-// Create request id span which increments source int each call. Capable of holding 8 digit number.
-static az_span get_request_id(void)
+/**************************************
+ Start JSON_BUILDER for JSON Document
+ This creates a new JSON with "{"
+*************************************/
+static az_result start_json_object(
+	az_json_writer* jw,
+	az_span az_span_buffer)
 {
-  az_span remainder;
-  az_span out_span = az_span_create((uint8_t*)request_id_buf, sizeof(request_id_buf));
-  az_result result = az_span_i32toa(out_span, request_id_int++, &remainder);
-  (void)remainder;
-  (void)result;
-  return out_span;
+	RETURN_ERR_IF_FAILED(az_json_writer_init(jw, az_span_buffer, NULL));
+	RETURN_ERR_IF_FAILED(az_json_writer_append_begin_object(jw));
+	return AZ_OK;
 }
 
-static int mqtt_publish_message(char* topic, az_span payload, int qos)
+/**********************************************
+* Start JSON_BUILDER for JSON Document
+* This adds "}" to the JSON
+**********************************************/
+static az_result end_json_object(
+	az_json_writer* jw)
 {
-  mqttPublishPacket cloudPublishPacket;
-  // Fixed header
-  cloudPublishPacket.publishHeaderFlags.duplicate = 0;
-  cloudPublishPacket.publishHeaderFlags.qos = qos;
-  cloudPublishPacket.publishHeaderFlags.retain = 0;
-  // Variable header
-  cloudPublishPacket.topic = (uint8_t*)topic;
-
-  // Payload
-  cloudPublishPacket.payload = az_span_ptr(payload);
-  cloudPublishPacket.payloadLength = az_span_size(payload);
-
-  if (MQTT_CreatePublishPacket(&cloudPublishPacket) != true)
-  {
-    debug_printError("MQTT: Connection lost PUBLISH failed");
-  }
-
-  return 0;
+	RETURN_ERR_IF_FAILED(az_json_writer_append_end_object(jw));
+	return AZ_OK;
 }
 
-// Send the response of the command invocation
-static int send_command_response(
-    az_iot_pnp_client_command_request* request,
-    uint16_t status,
-    az_span response)
+/**********************************************
+*	Add a JSON key-value pair with int32 data
+*	e.g. "property_name" : property_val (number)
+**********************************************/
+static az_result append_jason_property_int32(
```

Comment:
here and elsewhere. should be able to do a find all and replace
```suggestion
static az_result append_json_property_int32(
```